src/main/java/de/zalando/aruha/nakadi/service/EventPublisher.java

```diff
@@ -65,6 +74,20 @@ public EventPublishResult publish(final JSONArray events, final String eventType
         }
     }
 
+    private void enrich(final List<BatchItem> batch, final EventType eventType) throws EnrichmentException {
+        for (BatchItem item : batch) {
+            try {
+                item.setStep(EventPublishingStep.ENRICHING);
+                enrichment.enrich(item.getEvent(), eventType);
```

Comment:
Yes, I think there's no good reason to return anything as the method mutates the parameter you pass.